nav2_util/include/nav2_util/service_client.hpp

```diff
@@ -125,8 +125,9 @@ class ServiceClient
   /**
   * @brief Block until a service is available
   * @param timeout Maximum timeout to wait for, default infinite
+  * @return true if the client is ready
   */
-  void wait_for_service(const std::chrono::nanoseconds timeout = std::chrono::nanoseconds::max())
+  bool wait_for_service(const std::chrono::nanoseconds timeout = std::chrono::nanoseconds::max())
```

Comment:
Why a bool? It throws an exception if its not ready, there's no need to return the bool if success/failed because you'll know if an exception wasn't thrown

Reply:
The exception is thrown only if rclcpp is not ok.
And it might be better to remove the while loop and only keep wait_for_service(timeout)?
I don't really get the behavior with it.
https://github.com/ros-planning/navigation2/blob/ed43bed86253961f2723bb38389e51cabf6e7ebf/nav2_util/include/nav2_util/service_client.hpp#L132-L138
And using a return to get the result seems cleaner than using a try/catch.

Reply:
```
return client_->wait_for_service(timeout);
```
Should be sufficient?

Reply:
This is an infinite while loop, this will never return a non-successful state because it will block forever or throw an exception in a failure state.